target/linux/ramips/dts/mt7621_h3c_tx180x.dtsi

```diff
@@ -86,8 +95,13 @@
 		};
 
 		partition@80000 {
+			compatible = "u-boot,env";
 			label = "u-boot-env";
 			reg = <0x0080000 0x0080000>;
+
+			macaddr_uboot_eth: ethaddr {
+				#nvmem-cell-cells = <1>;
```

Comment:
Please search `package/boot/uboot-envtools/files/ramips` to find the "u-boot,env" size. If the partition size is not equal to the env size, we need to create a sub partition to wrap it. That's because the driver will check the u-env crc32 value. see: https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/drivers/nvmem/u-boot-env.c?h=next-20231208#n186

i.e.
https://github.com/openwrt/openwrt/pull/13120/commits/234bf01178731a882076af298f79764feeaedffd

Reply:
Ah right. I have a device like this. Unfortunate.

Reply:
I believe this is fine now.

Reply:
Although I do wonder if everything should be in a subnode for consistency...

Reply:
The data length of env on the following devices is ```0x1000```.

- ELECOM WRC-2533GHBK-I
- ELECOM WRC-2533GHBK2-T
- WNPR2600G

Reply:
Based on my reading of https://github.com/openwrt/openwrt/commit/db93d6ca67dbdbe1b496b53c9b85d777bbce448a, the size should be the last value of ubootenv_add_uci_config